src/pages/en/reference/integrations-reference.md

```diff
@@ -40,30 +39,159 @@ interface AstroIntegration {
+
+#### "updateConfig" option
+
+**Type:** `(newConfig: Record<string, any>) => void;`
+
+A callback function to update the user-supplied [Astro config](/en/reference/configuration-reference/). Any config you provide **will be merged with the user config,** so you are free to omit keys! 
+
+For example, say you need to supply a [Vite](https://vitejs.dev) plugin to the user's project:
+
+```js
+import bananaCSS from '@vitejs/official-banana-css-plugin';
+
+export default {
+  name: 'banana-css-integration',
+  hooks: {
+    'astro:config:setup': ({ updateConfig }) => {
+      updateConfig({
+        vite: {
+          plugins: [bananaCSS()],
+        }
+      })
+    }
+  }
+}
+```
+
+#### "addRenderer" option
+
+- **Type:** [`AstroRenderer`](https://github.com/withastro/astro/blob/fdd607c5755034edf262e7b275732519328a33b2/packages/astro/src/%40types/astro.ts#L872-L883)
+- **Examples:** [`lit`](https://github.com/withastro/astro/blob/main/packages/integrations/lit/src/index.ts), [`svelte`](https://github.com/withastro/astro/blob/main/packages/integrations/svelte/src/index.ts), [`react`](https://github.com/withastro/astro/blob/main/packages/integrations/react/src/index.ts), [`preact`](https://github.com/withastro/astro/blob/main/packages/integrations/preact/src/index.ts), [`vue`](https://github.com/withastro/astro/blob/main/packages/integrations/vue/src/index.ts), [`solid`](https://github.com/withastro/astro/blob/main/packages/integrations/solid/src/index.ts)
+
+A callback function to add a component framework renderer (i.e. React, Vue, Svelte, etc). You can browse the examples and type definition above for more advanced options, but here are the 2 main options to be aware of:
+- `clientEntrypoint` - path to a file that executes on the client whenever your component is used. This is mainly for rendering or hydrating your component with JS.
+- `serverEntrypoint` - path to a file that executes during server-side requests or static builds whenever your component is used. These should render components to static markup, with hooks for hydration where applicable. [React's `renderToString` callback](https://reactjs.org/docs/react-dom-server.html#rendertostring) is a classic example.
+
+#### "injectScript" option
+
+**Type:** `(stage: InjectedScriptStage, content: string) => void;`
+
+A callback function to inject a string of JavaScript content onto every page.
+
+The **`stage`** denotes how this script (the `content`) should be inserted. Some stages are inserted purely, while others are picked up by Vite:
+
+- `"before-hydration"`: Imported client-side, before the hydration script runs. Processed & resolved by Vite.
+- `"head-inline"`: Injected into a script tag in the `<head>` of every page. Not processed or resolved by Vite.
+- `"page"`: Injected into the JavaScript bundle of every page. Processed & resolved by Vite.
+- `"page-ssr"`: Injected into the frontmatter of every Astro page. Processed & resolved by Vite.
```

Comment:
Could I ask for a bit of clarity around this; `page-ssr` this inject code directly into every `src/page/**/*.astro` file's frontmatter section, if so does it add to the head or the tail of the block?

Reply:
Asking the team about this! I felt pretty uneasy writing this section, because I honestly don't know which I would choose and when. Definitely re-drafting and adding example use cases 👍

Reply:
Yea, agreed that these two sound the same here. Here's how I'd write them, @aFuzzyBear lmk if this is more clear:

```
- `"page"`: Similar to `head-inline`, except that the injected snippet is handled by Vite and bundled with any other `<script>` tags defined inside of Astro components on the page. The script will be loaded with a `<script type="module">` in the final page output.
- `"page-ssr"`: Injected into the frontmatter of every Astro page component. This is not commonly used, however it can be useful for injecting a CSS `import` into every page via its frontmatter.
```

Reply:
That is indeed alot clearer thank you guys 💚

Reply:
Makes sense to me too!